src/zarpaint/_interpolate_labels.py

```diff
@@ -0,0 +1,341 @@
+                    )
+            self.interp_dim = 0
+        else:
+            self.interp_dim = interp_dim
+
+    def enter_interpolation(self, event):
+        """Connect the paint callback and change button text
+
+        Parameters
+        ----------
+        event : Event
+            Event spawned by button click
+        """
+        self.selected_layer = self.viewer.layers[
+                self.labels_combo.current_choice]
+
+        self.selected_layer.events.paint.connect(self.paint_callback)
+
+        self.start_interpolation_btn.hide()
+        self.interpolate_btn.show()
+
+        self.interpolate_btn.clicked.connect(self.interpolate)
+
+    def interpolate(self, event):
+        """For each label_id, iterate over each slice that has been painted on
+         and perform pairwise (i, i+1) interpolation on each pair.
+
+        Parameters
+        ----------
+        event : Event
+            Object created upon clicking "interpolate" in the widget
+        """
+        assert self.interp_dim is not None, 'Cannot interpolate without knowing dimension'
+
+        for label_id, slices_painted in self.painted_slice_history.items():
+            slices_painted = list(sorted(slices_painted))
+            if len(slices_painted) > 1:
+                for i in range(1, len(slices_painted)):
+                    interpolate_between_slices(
+                            self.selected_layer, slices_painted[i - 1],
+                            slices_painted[i], label_id, self.interp_dim
+                            )
+
+        self.reset()
+
+    def reset(self):
+        """Reset button text and clear paint event history
+        """
+        self.selected_layer.events.paint.disconnect(self.paint_callback)
+        self.painted_slice_history.clear()
+        self.interp_dim = None
+        self.interpolate_btn.clicked.disconnect(self.interpolate)
+
+        self.interpolate_btn.hide()
+        self.start_interpolation_btn.show()
+
+
+def interpolate_between_slices(
+        label_layer: "napari.layers.Labels",
+        slice_index_1: int,
+        slice_index_2: int,
+        label_id: int = 1,
+        interp_dim: int = 0
+        ):
+    """Compute and draw interpolation between 2 label annotations.
+
+    Parameters
+    ----------
+    label_layer : napari.layers.Labels
+        The label layer to draw on 
+    slice_index_1 : int
+        slice containing the first label annotation
+    slice_index_2 : int
+        slice containing the second label anotation
+        interpolation occurs between slice_index_1 slice_index_2
+    label_id : int, optional
+        the id of the annotation that is to be painted, by default 1
+    interp_dim : int, optional
+        the dimension/axis to interpolate across, by default 0
+    """
+
+    if slice_index_1 > slice_index_2:
+        slice_index_1, slice_index_2 = slice_index_2, slice_index_1
+    layer_data = np.asarray(label_layer.data)
```

Comment:
This line needs to be removed. (I assume it's here because you were debugging something, but we can't have an entire full size volume read into memory).

We should replace the following two lines with:
```python
    slice_1 = np.take(label_layer.data, slice_index_1, axis=interp_dim)
    slice_2 = np.take(label_layer.data, slice_index_2, axis=interp_dim)
```